File: swagger/models/gift_card.py
from .base_model import BaseModel


class GiftCard(BaseModel):
	"""
	no description available

	Attributes:

		id
		The gift card's `ProductID`.

		location_ids
		The IDs of the locations where the gift card is sold.

		description
		A description of the gift card.

		editable_by_consumer
		When `true`, indicates that the gift card can be edited by the client.

		card_value
		The value of the gift card.

		sale_price
		The sale price of the gift card, if applicable.

		sold_online
		When `true`, indicates that the gift card is sold online.

		membership_restriction_ids
		A list of IDs for membership restrictions, if this card is restricted
		to members with certain types of memberships.

		gift_card_terms
		The terms of the gift card.

		contact_info
		Contact information for the gift card.

		display_logo
		When `true`, indicates that the logo should be displayed on the gift
		card.

		layouts
		A list of layouts available for the gift card.
	"""

	input_map = {
		'Id': 'id',
		'LocationIds': 'location_ids',
		'Description': 'description',
		'EditableByConsumer': 'editable_by_consumer',
		'CardValue': 'card_value',
		'SalePrice': 'sale_price',
		'SoldOnline': 'sold_online',
		'MembershipRestrictionIds': 'membership_restriction_ids',
		'GiftCardTerms': 'gift_card_terms',
		'ContactInfo': 'contact_info',
		'DisplayLogo': 'display_logo',
		'Layouts': 'layouts',
		}

	output_map = {
		'id': 'Id',
		'location_ids': 'LocationIds',
		'description': 'Description',
		'editable_by_consumer': 'EditableByConsumer',
		'card_value': 'CardValue',
		'sale_price': 'SalePrice',
		'sold_online': 'SoldOnline',
		'membership_restriction_ids': 'MembershipRestrictionIds',
		'gift_card_terms': 'GiftCardTerms',
		'contact_info': 'ContactInfo',
		'display_logo': 'DisplayLogo',
		'layouts': 'Layouts',
		}

File: swagger/Models/GiftCard.php
<?php

namespace App\Mindbody\Models;

/**
 * no description available
 */
class GiftCard extends BaseModel
{
	/**
	 * The gift card's `ProductID`.
	 * 
	 * @var integer format:int32
	 */
	public $id;
	/**
	 * The IDs of the locations where the gift card is sold.
	 * 
	 * @var integer[] 
	 */
	public $locationIds;
	/**
	 * A description of the gift card.
	 * 
	 * @var string 
	 */
	public $description;
	/**
	 * When `true`, indicates that the gift card can be edited by the client.
	 * 
	 * @var boolean 
	 */
	public $editableByConsumer;
	/**
	 * The value of the gift card.
	 * 
	 * @var number format:double
	 */
	public $cardValue;
	/**
	 * The sale price of the gift card, if applicable.
	 * 
	 * @var number format:double
	 */
	public $salePrice;
	/**
	 * When `true`, indicates that the gift card is sold online.
	 * 
	 * @var boolean 
	 */
	public $soldOnline;
	/**
	 * A list of IDs for membership restrictions, if this card is restricted
	 * to members with certain types of memberships.
	 * 
	 * @var integer[] 
	 */
	public $membershipRestrictionIds;
	/**
	 * The terms of the gift card.
	 * 
	 * @var string 
	 */
	public $giftCardTerms;
	/**
	 * Contact information for the gift card.
	 * 
	 * @var string 
	 */
	public $contactInfo;
	/**
	 * When `true`, indicates that the logo should be displayed on the gift
	 * card.
	 * 
	 * @var boolean 
	 */
	public $displayLogo;
	/**
	 * A list of layouts available for the gift card.
	 * 
	 * @var GiftCardLayout[] 
	 */
	public $layouts;

	protected function getInputMap()
	{
		return [
			'Id' => ['id', 'integer', 'int32'],
			'LocationIds' => ['locationIds', 'array', null],
			'Description' => ['description', 'string', null],
			'EditableByConsumer' => ['editableByConsumer', 'boolean', null],
			'CardValue' => ['cardValue', 'number', 'double'],
			'SalePrice' => ['salePrice', 'number', 'double'],
			'SoldOnline' => ['soldOnline', 'boolean', null],
			'MembershipRestrictionIds' => ['membershipRestrictionIds', 'array', null],
			'GiftCardTerms' => ['giftCardTerms', 'string', null],
			'ContactInfo' => ['contactInfo', 'string', null],
			'DisplayLogo' => ['displayLogo', 'boolean', null],
			'Layouts' => ['layouts', 'array', GiftCardLayout::class],
			];
	}

	protected function getOutputMap()
	{
		return [
			'id' => ['Id', 'integer', 'int32'],
			'locationIds' => ['LocationIds', 'array', null],
			'description' => ['Description', 'string', null],
			'editableByConsumer' => ['EditableByConsumer', 'boolean', null],
			'cardValue' => ['CardValue', 'number', 'double'],
			'salePrice' => ['SalePrice', 'number', 'double'],
			'soldOnline' => ['SoldOnline', 'boolean', null],
			'membershipRestrictionIds' => ['MembershipRestrictionIds', 'array', null],
			'giftCardTerms' => ['GiftCardTerms', 'string', null],
			'contactInfo' => ['ContactInfo', 'string', null],
			'displayLogo' => ['DisplayLogo', 'boolean', null],
			'layouts' => ['Layouts', 'array', GiftCardLayout::class],
			];
	}
}
